.circleci/config.yml

```diff
@@ -1,216 +1,33 @@
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-exception-spotbugs
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-exception-spoon:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-exception-spoon
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-exception-spring-framework:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-exception-spring-framework
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-exception-hbase:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-exception-hbase
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-exception-Pmd-elasticsearch-lombok-ast:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-exception-Pmd-elasticsearch-lombok-ast
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-exception-alot-of-projects:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-exception-alot-of-projects
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-warning-imports-guava:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-warning-imports-guava
-      - run:
-          command: ./.ci/validation.sh git-diff
-  no-warning-imports-java-design-patterns:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh no-warning-imports-java-design-patterns
-      - run:
-          command: ./.ci/validation.sh git-diff
-  sonarqube:
-    docker:
-      - image: *default-image
-    steps:
-      - checkout
-      - run:
-          command: |
-            ./.ci/validation.sh sonarqube
-      - run:
-          command: ./.ci/validation.sh git-diff
+commands:
+  run-script:
+    parameters:
+      script-name:
+        description: "name of script"
+        type: string
+        default: ""
+      args:
+        description: "arguments to script"
+        type: string
+        default: ""
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore Maven repo cache
+          keys:
+            - mvn-cache-{{ checksum "pom.xml" }}
+      - run:
+          command: << parameters.script-name >> << parameters.args >>
+      - run:
+          command: ./.ci/validation.sh git-diff
+      - save_cache:
+          name: Save Maven repo cache
+          key: mvn-cache-{{ checksum "pom.xml" }}
```

Comment:
Pointing to other conversation at https://github.com/checkstyle/checkstyle/pull/12365#discussion_r1014564736 and that whole PR ,

POM changes too often and we agreed to quarterly cache period. This should be updated to `(month - 1) / 3`.

Reply:
Quarterly is much too long, Circleci does not even allow a cache to live that long anyway. Max lifetime of a CircleCI cache is two weeks.

Reply:
Ok, then why don't we just make this a static field? We don't change that much in a 2 weeks period that we would grow our cache out of control.

Reply:
This allows us to identify a given cache in case of some issue (bad download, experimental dependency cached, dependabot PR, etc).

Reply:
Non of cache model of very good, it all depends ..., usage of any is good enough. 
Circle CI recommend this caching for maven, ok.